packages/editor/src/components/rich-text/index.js

```diff
@@ -426,7 +426,13 @@ export class RichText extends Component {
 			return;
 		}
 
-		const { start, end } = this.createRecord();
+		const { start, end, formats } = this.createRecord();
```

Comment:
`createRecord` creates a new record form the DOM. `getRecord` gets the current record from props. Also I think this might be better to do in onChange right before or after you update the global state. You have direct access to the record there. Thanks!

Reply:
looks like onChange only triggers when typing instead of when using cursor keys or changing selection with the mouse like I need.

The createRecord was there before, should it be getRecord instead?

Reply:
Oh, never mind then, I only saw the green line. :) Then this looks good, sorry!

Reply:
No worries - thanks for taking a look. 😄